apps/src/templates/progress/googlePlatformApiRedux.js

```diff
@@ -0,0 +1,98 @@
+import {getStore} from '@cdo/apps/redux';
+
+/*
+ * The Google Classroom Share Button is only available through the google
+ * platform api, so we have to add it to our page on load. We manage loading
+ * in redux since multiple components need to observe the loading state.
+ */
+
+const START_LOADING_GAPI = 'googlePlatformApi/START_LOADING_GAPI';
+const FINISH_LOADING_GAPI = 'googlePlatformApi/FINISH_LOADING_GAPI';
+
+export const startLoadingGapi = () => ({type: START_LOADING_GAPI});
+const finishLoadingGapi = success => ({type: FINISH_LOADING_GAPI, success});
+
+const GOOGLE_PLATFORM_API_ID = 'GooglePlatformApiId';
+const LOAD_TIMEOUT_MILLIS = 15000;
+
+const initialState = {
+  loading: false,
+  loaded: false,
+  loadStartTime: null
+};
+
+export function canShowGoogleShareButton(state) {
+  return !!state.googlePlatformApi && state.googlePlatformApi.loaded;
+}
+
+export default function googlePlatformApi(state = initialState, action) {
+  if (action.type === START_LOADING_GAPI) {
+    if (gapiReady()) {
+      return {
+        ...state,
+        loaded: true
+      };
+    } else {
+      loadApi();
```

Comment:
The reducer should only transform state and perform no side effects (other parts of progress redux have failed to follow this rule and have caused significant tech debt). These should be moved to the action or to the component that calls it.
https://redux.js.org/basics/reducers (Search for "Things you should never do inside a reducer").

Reply:
Some examples that may be helpful: https://github.com/code-dot-org/code-dot-org/blob/staging/apps/src/code-studio/components/shareDialogRedux.js
https://redux.js.org/advanced/example-reddit-api

Reply:
thanks for those examples. doing that stuff in the reducers didn't feel right, but again it was too easy to find examples in our code that made it seem okay. i will update!